constantine/commitments/ipa/test_helper.nim

```diff
@@ -0,0 +1,248 @@
+# Constantine
+# Copyright (c) 2018-2019    Status Research & Development GmbH
+# Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+# Licensed and distributed under either of
+#   * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+#   * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+import 
+    ./[multiproof, barycentric_form],
+    ./helper_types,
+    ../../../constantine/math/config/[type_ff, curves],
+    ../../../constantine/math/elliptic/[
+     ec_twistededwards_affine,
+     ec_twistededwards_projective
+      ],
+    ../../../constantine/math/io/io_fields,
+    ../../../constantine/serialization/[
+      codecs_status_codes,
+      codecs_banderwagon,
+      codecs
+       ],
+    ../../../constantine/math/arithmetic
+
+# ############################################################
+#
+#       All the helper functions required for testing
+#
+# ############################################################
+
+func evaluate* [EC_P_Fr] (res: var EC_P_Fr, poly: openArray[EC_P_Fr], point: EC_P_Fr,  n: static int) = 
```

Comment:
Same remark as previously, use a generic `Field` or just `Fr`, probably need an ipa prefix aswell so we can easily distinguish between IPA, KZG, Polynomial evaluation when reviewing / auditing code.
```suggestion
func ipaEvaluate*(res: var Fr, poly: openArray[Fr], point: Fr,  n: static int) = 
```

Reply:
Also what is ``n`` for?

Reply:
ipaEvaluate would work on arrays of different lengths, on different rounds, so decided to pass the length along.